src/PrestaShopBundle/Model/Product/AdminModelAdapter.php

```diff
@@ -550,7 +551,7 @@ private function getVirtualProductData()
 
         $id_product_download = \ProductDownloadCore::getIdFromIdProduct((int)$this->product->id, false);
```

Comment:
`ProductDownloadCore` => `ProductDownload`

Maybe trying a regex checking static properties would help you ?